### PR TITLE
Tier-3 batch 1: Euclid, AbsDiff, Midpoint, From<u64>

### PR DESCRIPTION
**Pull request by kaidokert**
First batch of the Tier-3 convenience polish, mirroring FixedUInt.

- **Euclid / CheckedEuclid** (Nct) — unsigned Euclidean div/rem is ordinary div/rem; `checked_*` return `None` on a zero divisor.
- **AbsDiff** (Nct) — `|a - b|` = larger − smaller. Nct-only: the compare-branch isn't constant-time (FixedUInt's Ct arm uses a branchless select; heapless keeps its Ct trait surface minimal, same as saturating).
- **Midpoint** (personality-generic) — `(a & b) + ((a ^ b) >> 1)`, branchless (so CT-safe) and width-preserving (single-bit shift keeps `len`; `&`/`^`/`+` resolve at `max(len)`).
- **From<u64>** — extend the `from_primitive` macro; source-int width like the others.

Tests: generic `for_both_carriers` body (Euclid/abs_diff/midpoint, incl. the midpoint no-overflow `max` case) plus a `From<u64>` source-width check. Verified default / nightly / --no-default-features; clippy clean.

First of a few Tier-3 batches — power-of-two/multiples, roots/logs, and iterators/bytes/CtParity follow.

## Summary by Sourcery

Add Euclid, AbsDiff, and Midpoint trait support for heapless big integers and extend primitive conversions to include u64, with corresponding tests and documentation updates.

New Features:
- Implement Euclid and CheckedEuclid for HeaplessBigInt in the non-constant-time personality.
- Implement AbsDiff for HeaplessBigInt in the non-constant-time personality.
- Implement personality-generic Midpoint for HeaplessBigInt.
- Extend HeaplessBigInt From<uN> support to include u64, preserving source integer width.

Documentation:
- Update heapless module documentation to describe From<u8/u16/u32/u64> behavior and source-int width semantics.

Tests:
- Add generic carrier tests covering Euclid, CheckedEuclid, AbsDiff, and Midpoint behavior, including midpoint no-overflow max case.
- Add a HeaplessBigInt From<u64> test verifying limb count and layout for source-width construction.

## Summary by CodeRabbit

* **New Features**
  * Added Euclidean division, remainder, checked arithmetic, absolute difference, and midpoint operations for heapless big integers.
  * Added support for constructing heapless big integers from `u64` values.

* **Bug Fixes**
  * Checked Euclidean operations now safely return no result when dividing by zero.

* **Tests**
  * Added coverage for the new arithmetic operations and `u64` conversions across supported integer types.

### CODE REVIEW

File: src/heapless/euclid.rs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            Some(self % v)
+        }
+    }
+    fn checked_div_rem_euclid(self, v: Self) -> Option<(Self, Self)> {
+        if <Self as Zero>::is_zero(&v) {
+            None
+        } else {
+            Some(self.div_rem(&v))
+        }
+    }
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply src/heapless/euclid.rs:44]: @gemini-code-assist Done in 66396f1 — checked_div_euclid/checked_rem_euclid now delegate to the inherent checked_div/checked_rem, dropping the duplicated zero guard. checked_div_rem_euclid keeps a single is_zero() check since there's no inherent checked_div_rem.

File: tests/carrier_generic.rs [hunk 2]
```diff
@@ -458,3 +463,49 @@ fn pow() {
...
+        // midpoint averages without overflow.
+        assert_eq!(
+            Midpoint::midpoint(C::from_u32(10), C::from_u32(20)),
+            C::from_u32(15)
+        );
+        let max = C::from_u32(MAX32);
+        assert_eq!(Midpoint::midpoint(max, max), max); // (max + max) / 2 = max
+    }
+    for_both_carriers!(body);
+}
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply tests/carrier_generic.rs:524]: Added in e1c55e1 — checked_rem_euclid(x, 0) and both checked_div_rem_euclid paths (nonzero + zero divisor) are now asserted.
